### PR TITLE
[v14] Fix Assume Roles switch back, don't delete role if access list is using it.

### PR DESCRIPTION
**Pull request by mdwn**
Backport https://github.com/gravitational/teleport/pull/33746 to branch/v14.

### CODE REVIEW

File: lib/auth/access.go [hunk 1]
```diff
@@ -77,8 +85,45 @@ func (a *Server) DeleteRole(ctx context.Context, name string) error {
...
 				// within the system.
 				log.Warnf("Failed to delete role: role %v is used by user cert authority %v", name, a.GetClusterName())
-				return trace.BadParameter("failed to delete a role that is still in use by a user, check the system server logs for more details")
+				return trace.Wrap(errDeleteRoleCA)
+			}
+		}
+	}
+
+	var nextToken string
+	for {
```
Comment (r0mant) [inline lib/auth/access.go:94]: Is there no "forEach" kind of helper method for access lists already somewhere?

File: lib/auth/access.go [hunk 2]
```diff
@@ -77,8 +85,45 @@ func (a *Server) DeleteRole(ctx context.Context, name string) error {
...
+		var accessLists []*accesslist.AccessList
+		var err error
+		accessLists, nextToken, err = a.Services.AccessListClient().ListAccessLists(ctx, 0 /* default page size */, nextToken)
+		if err != nil {
+			return trace.Wrap(err)
+		}
+
+		for _, accessList := range accessLists {
+			for _, r := range accessList.Spec.Grants.Roles {
```
Comment (r0mant) [inline lib/auth/access.go:103]: Nit: I think we have a "contains" method for string slices somewhere in utils which will make this code a bit cleaner and avoid 3rd nested for loop.

File: lib/auth/access.go (hunk 1 above)
Comment (mdwn) [reply lib/auth/access.go:94]: No, unfortunately. :-(

File: lib/auth/access.go (hunk 2 above)
Comment (mdwn) [reply lib/auth/access.go:103]: I've elected to use `slices` instead here, let me know what you think.

**Review by r0mant — APPROVED** [review]

File: lib/auth/access.go (hunk 2 above)
Comment (r0mant) [reply lib/auth/access.go]: Even better.

Comment (camscale) [reply lib/auth/access.go]: Just dropping this here: https://github.com/gravitational/teleport/pull/33851 - this is the equivalent slices change on master that keeps master and branch/v14 in sync.

**Review by camscale — APPROVED** [review]

**Pull request merged**